routes/api.js

```diff
@@ -867,41 +871,43 @@ module.exports = function(config) {
     function flattenKeys(keys, prefix, numDocs) {
         var result = [];
 
-        for(var i=0; i<keys.sorted_keys.length; i++) {
-            var hasNestedKeys;
-            if (keys.keys[keys.sorted_keys[i]].keys == null) {
-                hasNestedKeys = false;
-            }
-            else {
-                for(var key in keys.keys[keys.sorted_keys[i]].keys) {
-                    hasNestedKeys = true;
-                    break;
-                }
-            }
-
-            // Save the field if it's a primitive (at least once) or if it's an empty object
-            if ((keys.keys[keys.sorted_keys[i]].primitiveValueCount > 0) || (hasNestedKeys === false)) {
-                result.push(prefix.concat(keys.sorted_keys[i]))
-            }
-            else {
-                var count = 0;
-                for(var type in keys.keys[keys.sorted_keys[i]].type) {
-                    count += keys.keys[keys.sorted_keys[i]].type[type]
-                }
-                if (typeof keys.keys[keys.sorted_keys[i]].primitiveCount === 'number') {
-                    count += keys.keys[keys.sorted_keys[i]].primitiveCount;
-                }
-
-                if (count < numDocs) {
-                    result.push(prefix.concat(keys.sorted_keys[i]))
-                }
-            }
-
-            // Recursively add more fields
-            if ((keys.keys[keys.sorted_keys[i]].type.object != null)
-                && (keys.keys[keys.sorted_keys[i]].type.object > 0)) {
-                result.push.apply(result, flattenKeys(keys.keys[keys.sorted_keys[i]], prefix.concat([keys.sorted_keys[i]]), numDocs))
-            }
+        if(keys.sorted_keys){
```

Comment:
One of the errors, where keys.sorted_keys is undefined, as the some of the keys in the document, are guids.  Just to unblock on the error, added the check for keys.sorted_keys. an example document.
[{"apiKeys":{"ZfVcjeBJlVrV1pA3KmNU4ThI3Brtz3n81WAtJBouGs8=":{"algo":"RS256","key":{"$reql_type$":"BINARY","data":"TmlLOUwKR0VONitKU1dmZ2NrUUNzNlVVQk9YU1pkcmVOTjl6YlFDd3l6ZWU3Yk9KcVhVREF1TGNG"
}}},
"encryptionKey":{"algo":"RS256","key":{"$reql_type$":"BINARY","data":"QVJ6UHcxRXNaQXlqVgp0R0NLSVEwL2J0cUsrakZ1blQyTkJDOFJJdGFuRFpwcHRRSURBUUFCCi0t"}},
"id":"test","owner":"testowner","tenant":"test_tenant"}]